### PR TITLE
implementing tests for the keyer

### PR DESCRIPTION
**Pull request by arschles**
Fixes https://github.com/kubernetes-incubator/service-catalog/issues/578

### CODE REVIEW

File: pkg/storage/tpr/keyer_test.go [hunk 1]
```diff
@@ -18,16 +18,67 @@ package tpr
...
+	ctx = api.WithNamespace(ctx, ctxNS)
+	keyer := Keyer{Separator: separator, ResourceName: resourceName}
+	key := keyer.Key(ctx, resourceName)
+	expected := ctxNS + separator + resourceName
+	if key != expected {
+		t.Fatalf("key was '%s', not expected '%s', key, expected")
+	}
 }
 
 func TestNamespaceAndNameFromKey(t *testing.T) {
```
Comment (simonleung8) [inline pkg/storage/tpr/keyer_test.go]: I think we are missing the test case of `key` being invalid.

Comment (krancour) [inline pkg/storage/tpr/keyer_test.go]: Add this and then lgtm.

**Comment by arschles** [comment]
@krancour @simonleung8 invalid key tests added

**Comment by simonleung8** [comment]
@arschles Thanks, lgtm now.

**Comment by arschles** [comment]
The latest build appears to be a flake. I've captured it in https://github.com/kubernetes-incubator/service-catalog/issues/643

**Pull request merged**